ntbs-service/Pages/Notifications/NotificationEditModelBase.cs

```diff
@@ -110,6 +109,12 @@ public async Task<IActionResult> OnPostAsync(bool isBeingSubmitted)
             }
         }
 
+        private bool UserHasSharePermissionsAndOnCorrectPage()
```

Comment:
It's a little weird having subclass specific logic in the base class, but if it simplifies the code structure then it's probably worth it